<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>UISettingsConnection</class>
 <widget class="QWidget" name="UISettingsConnection">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>691</width>
    <height>421</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="1" column="0">
    <widget class="QTabWidget" name="tabWidget">
     <property name="toolTip">
      <string/>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>Connection</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>Incomming connections</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_5">
          <property name="margin">
           <number>1</number>
          </property>
          <item row="0" column="0">
           <widget class="QWidget" name="widget" native="true">
            <layout class="QGridLayout" name="gridLayout_2">
             <property name="margin">
              <number>0</number>
             </property>
             <item row="1" column="0">
              <widget class="QRadioButton" name="radioButton_ACTIVE">
               <property name="toolTip">
                <string>Select this mode if your computer is connected directly to the internet.
Do not select this mode if you have a router or any external firewall.
If your computer is not behind a router all you have to do is to choose
this option and usually you don't need to alter anything more in the
connection settings.
However, if you have a software firewall installed, you may need further configuration in the firewall to allow Internet access for EiskaltDC++.</string>
               </property>
               <property name="text">
                <string>Active mode</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="QRadioButton" name="radioButton_PORT">
               <property name="toolTip">
                <string>Select this mode if you are behind a firewall (router), and will set up
the TCP, UDP and TLS port forwarding on your own.
You will need to specify the external IP address of your router using
the setting below if you want downloads to work properly.
Use this mode only if the previous ones are not options for you.</string>
               </property>
               <property name="text">
                <string>Firewall with port forwarding</string>
               </property>
               <property name="autoExclusive">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item row="5" column="0">
              <widget class="QRadioButton" name="radioButton_PASSIVE">
               <property name="toolTip">
                <string>Select this mode if none of the previous modes apply. In this mode,
EiskaltDC++ will not try to make inbound connections, only outbound ones.
This mode carries some serious limitations:
  * only 5 results per user will be returned in response to searches;
  * you will not be able to download from other passive users;
  * you may won't be able to search and/or download on certain hubs.</string>
               </property>
               <property name="text">
                <string>Passive mode</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QRadioButton" name="radioButton_UPNP">
               <property name="toolTip">
                <string>Select this mode if you are behind a firewall/router that supports
the UPnP standard for program auto-configuration.
Most routers do support UPnP but you may need to enable it
in the router's configuration page.
This way EiskaltDC++ will automatically open the ports it needs
and obtain your external IP address as well.</string>
               </property>
               <property name="text">
                <string>Firewall with UPnP</string>
               </property>
               <property name="checkable">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QFrame" name="frame">
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QGridLayout" name="gridLayout_3">
             <property name="margin">
              <number>0</number>
             </property>
             <item row="1" column="0">
              <widget class="QLabel" name="label_4">
               <property name="text">
                <string>External/WAN IP:</string>
               </property>
              </widget>
             </item>
             <item row="1" column="2" colspan="4">
              <widget class="LineEdit" name="lineEdit_WANIP">
               <property name="inputMask">
                <string>000.000.000.000; </string>
               </property>
              </widget>
             </item>
             <item row="0" column="0" colspan="6">
              <layout class="QGridLayout" name="gridLayout_4">
               <item row="0" column="0">
                <widget class="QLabel" name="label">
                 <property name="text">
                  <string>TCP port</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="label_2">
                 <property name="text">
                  <string>UDP port</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="label_3">
                 <property name="text">
                  <string>TLS port</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QSpinBox" name="spinBox_TCP">
                 <property name="minimum">
                  <number>1</number>
                 </property>
                 <property name="maximum">
                  <number>65535</number>
                 </property>
                 <property name="value">
                  <number>3030</number>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QSpinBox" name="spinBox_UDP">
                 <property name="minimum">
                  <number>1</number>
                 </property>
                 <property name="maximum">
                  <number>65535</number>
                 </property>
                 <property name="value">
                  <number>3030</number>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QSpinBox" name="spinBox_TLS">
                 <property name="minimum">
                  <number>1</number>
                 </property>
                 <property name="maximum">
                  <number>65535</number>
                 </property>
                 <property name="value">
                  <number>3031</number>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item row="3" column="0" colspan="6">
              <widget class="QCheckBox" name="checkBox_DONTOVERRIDE">
               <property name="text">
                <string>Don't allow Hub/UPnP to override</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>Outgoing connections</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_7">
          <property name="margin">
           <number>1</number>
          </property>
          <item row="0" column="0">
           <widget class="QWidget" name="widget_2" native="true">
            <layout class="QGridLayout" name="gridLayout_6">
             <property name="margin">
              <number>0</number>
             </property>
             <item row="1" column="0">
              <widget class="QRadioButton" name="radioButton_DC">
               <property name="toolTip">
                <string>Select this mode if you can make unrestricted outbound connections.
This includes connecting to any IP address on any port.
Choose this option if unsure.</string>
               </property>
               <property name="text">
                <string>Direct connection</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QRadioButton" name="radioButton_SOCKS">
               <property name="toolTip">
                <string>Select this mode if you want to make outbound connections through
a SOCKS proxy. The caveats for Passive mode also apply here.
SOCKS5 proxies are not the same as HTTP proxies.
DC++'s SOCKS5 support does not include GSSAPI.</string>
               </property>
               <property name="text">
                <string>SOCKS5</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QFrame" name="frame_2">
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QGridLayout" name="gridLayout_8">
             <property name="margin">
              <number>0</number>
             </property>
             <item row="0" column="1">
              <widget class="LineEdit" name="lineEdit_SIP">
               <property name="inputMask">
                <string>000.000.000.000; </string>
               </property>
              </widget>
             </item>
             <item row="2" column="1" colspan="3">
              <widget class="LineEdit" name="lineEdit_SUSR"/>
             </item>
             <item row="3" column="1" colspan="3">
              <widget class="LineEdit" name="lineEdit_SPSWD"/>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="label_5">
               <property name="text">
                <string>IP</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="label_7">
               <property name="text">
                <string>User</string>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="QLabel" name="label_8">
               <property name="text">
                <string>Password</string>
               </property>
              </widget>
             </item>
             <item row="0" column="2">
              <widget class="QLabel" name="label_6">
               <property name="text">
                <string>Port</string>
               </property>
              </widget>
             </item>
             <item row="0" column="3">
              <widget class="LineEdit" name="lineEdit_SPORT">
               <property name="minimumSize">
                <size>
                 <width>60</width>
                 <height>0</height>
                </size>
               </property>
               <property name="inputMask">
                <string>00000; </string>
               </property>
              </widget>
             </item>
             <item row="4" column="0" colspan="4">
              <widget class="QCheckBox" name="checkBox_RESOLVE">
               <property name="text">
                <string>Use to resolve hostnames</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>6</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Limits</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QGroupBox" name="groupBox_3">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="toolTip">
          <string/>
         </property>
         <property name="title">
          <string>Limits</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QCheckBox" name="checkBox_THROTTLE_ENABLE">
            <property name="text">
             <string>Enable total speed limit</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="frame_3">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <property name="margin">
              <number>0</number>
             </property>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_2">
               <property name="leftMargin">
                <number>10</number>
               </property>
               <property name="rightMargin">
                <number>10</number>
               </property>
               <item>
                <widget class="QLabel" name="label_11">
                 <property name="text">
                  <string>Download</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QSpinBox" name="spinBox_DOWN_LIMIT_NORMAL">
                 <property name="maximum">
                  <number>125000</number>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="label_14">
                 <property name="text">
                  <string>KiB/s</string>
                 </property>
                </widget>
               </item>
               <item>
                <spacer name="horizontalSpacer_2">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>40</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
               <item>
                <widget class="QLabel" name="label_12">
                 <property name="text">
                  <string>Upload</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QSpinBox" name="spinBox_UP_LIMIT_NORMAL">
                 <property name="maximum">
                  <number>125000</number>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="label_15">
                 <property name="text">
                  <string>KiB/s</string>
                 </property>
                </widget>
               </item>
               <item>
                <spacer name="horizontalSpacer_6">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>40</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_3">
               <property name="leftMargin">
                <number>10</number>
               </property>
               <item>
                <widget class="QCheckBox" name="checkBox_TIME_DEPENDENT_THROTTLE">
                 <property name="text">
                  <string>Alternate speed limit at time from </string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QFrame" name="frame_5">
                 <property name="frameShape">
                  <enum>QFrame::NoFrame</enum>
                 </property>
                 <property name="frameShadow">
                  <enum>QFrame::Raised</enum>
                 </property>
                 <layout class="QHBoxLayout" name="horizontalLayout_4">
                  <item>
                   <widget class="QSpinBox" name="spinBox_BANDWIDTH_LIMIT_START">
                    <property name="maximum">
                     <number>23</number>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="label_13">
                    <property name="text">
                     <string>to</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QSpinBox" name="spinBox_BANDWIDTH_LIMIT_END">
                    <property name="maximum">
                     <number>23</number>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="label_19">
                    <property name="text">
                     <string>hour(s)</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <spacer name="horizontalSpacer_4">
                    <property name="orientation">
                     <enum>Qt::Horizontal</enum>
                    </property>
                    <property name="sizeHint" stdset="0">
                     <size>
                      <width>40</width>
                      <height>20</height>
                     </size>
                    </property>
                   </spacer>
                  </item>
                 </layout>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QFrame" name="frame_4">
               <property name="frameShape">
                <enum>QFrame::NoFrame</enum>
               </property>
               <property name="frameShadow">
                <enum>QFrame::Raised</enum>
               </property>
               <layout class="QVBoxLayout" name="verticalLayout_5">
                <property name="margin">
                 <number>0</number>
                </property>
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout">
                  <item>
                   <widget class="QLabel" name="label_9">
                    <property name="text">
                     <string>Download</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QSpinBox" name="spinBox_DOWN_LIMIT_TIME">
                    <property name="maximum">
                     <number>125000</number>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="label_16">
                    <property name="text">
                     <string>KiB/s</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <spacer name="horizontalSpacer">
                    <property name="orientation">
                     <enum>Qt::Horizontal</enum>
                    </property>
                    <property name="sizeHint" stdset="0">
                     <size>
                      <width>40</width>
                      <height>20</height>
                     </size>
                    </property>
                   </spacer>
                  </item>
                  <item>
                   <widget class="QLabel" name="label_10">
                    <property name="text">
                     <string>Upload</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QSpinBox" name="spinBox_UP_LIMIT_TIME">
                    <property name="maximum">
                     <number>125000</number>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLabel" name="label_17">
                    <property name="text">
                     <string>KiB/s</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <spacer name="horizontalSpacer_5">
                    <property name="orientation">
                     <enum>Qt::Horizontal</enum>
                    </property>
                    <property name="sizeHint" stdset="0">
                     <size>
                      <width>40</width>
                      <height>20</height>
                     </size>
                    </property>
                   </spacer>
                  </item>
                 </layout>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>LineEdit</class>
   <extends>QLineEdit</extends>
   <header>LineEdit.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
